File: packages/cli/test/fixtures/ranking-corpora/swift/Sources/UpdateHandler.swift
import Foundation

/// Changes an existing user's display name.
struct UpdateHandler {
    func updateUser(_ id: Int, _ name: String) {
        if id <= 0 {
            Logger.logError("updateUser", NSError(domain: "id must be positive", code: 1))
            return
        }
        Logger.logInfo("updated user \(name)")
    }
}

File: packages/cli/test/fixtures/ranking-corpora/csharp/UpdateHandler.cs
namespace RankingCSharp;

/// <summary>Changes an existing user's display name.</summary>
public class UpdateHandler
{
    public void UpdateUser(int id, string name)
    {
        if (id <= 0)
        {
            Logger.LogError("UpdateUser", new ArgumentException("id must be positive"));
            return;
        }

        Logger.LogInfo($"updated user {name}");
    }
}
